ext/zend_test/zendmm_observer.c

```diff
@@ -0,0 +1,117 @@
+/*
+  +----------------------------------------------------------------------+
+  | Copyright (c) The PHP Group                                          |
+  +----------------------------------------------------------------------+
+  | This source file is subject to version 3.01 of the PHP license,      |
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | https://www.php.net/license/3_01.txt                                 |
+  | If you did not receive a copy of the PHP license and are unable to   |
+  | obtain it through the world-wide-web, please send a note to          |
+  | license@php.net so we can mail you a copy immediately.               |
+  +----------------------------------------------------------------------+
+  | Author:                                                              |
+  +----------------------------------------------------------------------+
+*/
+
+#include "php.h"
+#include "php_test.h"
+#include "Zend/zend_alloc.h"
+#include "ext/standard/info.h"
+#include "ext/standard/php_var.h"
+#include "zendmm_observer.h"
+
+void observer_malloc(size_t len, void *ptr ZEND_FILE_LINE_DC ZEND_FILE_LINE_ORIG_DC)
+{
+	size_t block_len = zend_mm_block_size(zend_mm_get_heap(), ptr);
```

Comment:
This breaks with `USE_ZEND_ALLOC=0` because `ptr` was allocated by `malloc`. I would suggest setting `block_len` to `0` when `!is_zend_ptr()` here